api.bs

```diff
@@ -97,23 +98,47 @@ New additions to the
 
 TODO explain why we use histograms
 
+* Compatibility with privacy-preserving aggregation systems
+* Flexibility to assign buckets
+
+* As histogram size increases, noise becomes a problem
+
 
 # Overview of Operation # {#overview}
 
-At impression time, information about an advertisement is saved by the browser in a write-only store.
-This includes an identifier for the ad and some metadata about the ad,
-such as whether the impression was an ad view or an ad click.
+The private attribution API provides aggregate information about the
+association between two classes of events: [=impressions=] and [=conversions=].
+
+An <dfn>impression</dfn>, sometimes called a *source event*, is the
+event to which [=conversion=]s are being attributed. Selection of impression
+events is left to the consumer of the API. Examples include:
+
+*   Displaying an advertisement to a user.
+*   Viewing a particular web page.
+
+A <dfn>conversion</dfn>, sometimes called a *trigger event*, is the
+event being attributed to [=impression=]s. Selection of conversion events
+is again left to the consumer of the API. Examples include:
+
+*   Signing up for an account.
+*   Making a purchase.
 
-At conversion time, information for aggregation is created based on the impressions that were previously stored.
-A site can request that the browser select impressions based on a simple query.
+When an [=impression=] occurs, information about the impression is saved by the
+browser in a write-only store. This includes an identifier for the impression
+and some metadata about the impression, such as whether the impression was an
+ad view or an ad click.
```

Comment:
```suggestion
ad view or an ad click.
```

The write-only piece is negotiable.  I can see why a site couldn't modify its store of impressions, but it's also not really necessary if the reading *by other sites* is controlled properly.

The whole business with "what does the e-Privacy Directive say about storing information" is silly.  It matters, of course, but not every regulatory system will grow a PECR-like piece of baggage.

Reply:
The `write-only` term was actually in the original and I also had some doubts about it. I think it's an oversimplification. We should discuss the characteristics of the store somewhere (I think I added a section for that), but it takes more than just saying "write only" to capture the important characteristics.